### ISSUE TITLE
Map cannot be scaled in scene.

### ISSUE THREAD

**Issue by ArslanJafri** [wrld3d/unity-api#11]
Currently scaling or repositioning the map in scene before running has no effect as child objects are dynamically created and do not match the scale or position of WRLD Map object. I was able to fix that with the following changes, it would be great if these or something similar could be included in the SDK.

Change CreateGameObject method in GameObjectFactory.cs to
```csharp
        private GameObject CreateGameObject(Mesh mesh, Material material, string objectName, bool createCollisionMesh)
        {
            var gameObject = new GameObject(objectName);
            gameObject.SetActive(false);
            gameObject.transform.parent = m_parentTransform;

            //Match parents scale in case Map has been scaled
            gameObject.transform.localScale = Vector3.one;

            gameObject.AddComponent<MeshFilter>().sharedMesh = mesh;
            gameObject.AddComponent<MeshRenderer>().sharedMaterial = material;
            if (createCollisionMesh)
            { 
                gameObject.AddComponent<MeshCollider>().sharedMesh = mesh;
            }
            return gameObject;
        }
```

and Constructor in GameObjectRepository.cs to

```csharp
        public GameObjectRepository(string rootName, Transform parentForStreamedObjects, MaterialRepository materialRepository)
        {
            m_root = new GameObject(rootName);
            m_root.transform.parent = parentForStreamedObjects;
            
            //Match parent's scale in case Map has been scaled
            m_root.transform.localScale = Vector3.one;
            //Match parents position in case Map has been moved
            m_root.transform.localPosition = Vector3.zero;

            m_materialRepository = materialRepository;
        }
```

**Comment by boblaublaw** [comment]
Was a fix for this put into SDK 0.5? I am on SDK 0.5 and map scaling is causing many problems for me as well.

**Comment by ghost** [comment]
This is now fixed.
